stdlib/public/Cxx/CxxDictionary.swift

```diff
@@ -144,6 +144,58 @@ extension CxxDictionary {
     }
   }
 
+  public func mapValues<R, E>(
+    _ transform: (Value) throws(E) -> R.Value
+  ) throws(E) -> R where R: CxxDictionary, R.Key == Key {
+    var result = R.init()
+
+    var iterator = __beginUnsafe()
+    let endIterator = __endUnsafe()
+
+    while iterator != endIterator {
+      let pair = iterator.pointee
+      try result.__insertUnsafe(R.Element(first: pair.first, second: transform(pair.second)))
+      iterator = iterator.successor()
+    }
+
+    return result
+  }
+
+  @inlinable
+  @_disfavoredOverload
```

Comment:
Just trying to understand this better, why do we need `@_disfavoredOverload` on this overload?